pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1479,8 +1479,11 @@ func (cf *changeFrontier) Start(ctx context.Context) {
 
 	if err := checkpoint.Restore(cf.frontier, cf.spec.SpanLevelCheckpoint); err != nil {
 		if log.V(2) {
-			log.Dev.Infof(cf.Ctx(), "change frontier encountered error on checkpoint restore: %v", err)
+			log.Dev.Infof(cf.Ctx(),
+				"change frontier moving to draining due to error restoring span-level checkpoint: %v", err)
```

Comment:
Is there any other info to log that might make it easier to debug in the future? E.g., which checkpoint is potentially problematic (may only be relevant for per-table checkpoints).

Reply:
I modified `checkpoint.Restore` so that it wraps the error with the specific (span, timestamp) pair that failed to be forwarded. We could potentially log the entire checkpoint but I'm leaning towards not doing that for now because it's likely to be large and if the problem is persistent, we could always decode it from the job_info table.